lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2508,6 +2516,116 @@ signTransaction tl preferredLatestEra keyLookup (rootKey, rootPwd) utxo =
             keyLookup
             inputResolver
 
+type MakeRewardAccountBuilder k =
+    (k 'RootK XPrv, Passphrase "encryption") -> (XPrv, Passphrase "encryption")
+
+-- | Produce witnesses and construct a transaction from a given selection.
+--
+-- Requires the encryption passphrase in order to decrypt the root private key.
+-- Note that this doesn't broadcast the transaction to the network. In order to
+-- do so, use 'submitTx'.
+--
+buildAndSignTransactionNoSelection
+    :: forall (n :: NetworkDiscriminant)
+     . (Typeable n, DelegationAddress n ShelleyKey 'CredFromKeyK)
```

Comment:
`ShelleyKey` won't work with `postTransactionOld`. Though `postTransactionOld` is to be tackled in a separate task, I'd advocate for keeping the type-signatures of `buildAndSignTransactionNoSelection` and `buildAndSignTransaction` as close as possible.